tests/test_validation.py

```diff
@@ -108,3 +111,55 @@ def test_wildcard_match(simple_df):
 
     with pytest.raises(ValueError, match=MATCH_FAIL_VALIDATION):
         definition.validate(simple_df.rename(scenario={"scen_a": "foo"}))
+
+
+@pytest.mark.parametrize(
+    "subannual, expected_error",
+    [
+        ("01-01 00:00+01:00", False),
+        ("01-01 00:00", True),
+        ("01-01 00:00+02:00", True),
+        ("01-32 00:00+01:00", True),
+    ],
+)
+def test_validate_datetime(simple_df, subannual, expected_error):
+    definition = DataStructureDefinition(
+        TEST_DATA_DIR / "data_structure_definition" / "subannual"
+    )
+    definition.config.time = nomenclature.config.TimeConfig(
+        year=True,
+        datetime="UTC+01:00",
+    )
+    df = IamDataFrame(simple_df._data, subannual=subannual)
+    if not expected_error:
+        assert definition.validate_datetime(df) is None
+    else:
+        with pytest.raises(ValueError):
+            definition.validate_datetime(df)
```

Comment:
Please remove this test case or change to only use the examples like months shown above. Using truncated datetime-strings in the subannual column was an hacky quickfix and shouldn't be supported in the future.